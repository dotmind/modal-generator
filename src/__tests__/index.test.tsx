/**
 * @jest-environment jsdom
*/

import React from 'react';
import { render } from '@testing-library/react';

import Modal from '..';

describe('React Modal', () => {
  it('Check if modal is in the document', () => {
    const { baseElement, queryByText } = render(<Modal modalOpen={true} onClose={() => { return; }}>modal content</Modal>);

    expect(baseElement).toMatchSnapshot();
    expect(queryByText('modal content')).toBeInTheDocument();
  });
  it('Check if modal is not in the document', () => {
    const { baseElement, queryByText } = render(<Modal modalOpen={false} onClose={() => { return; }}>modal content</Modal>);

    expect(baseElement).toMatchSnapshot();
    expect(queryByText('modal content')).not.toBeInTheDocument();
  });
});
