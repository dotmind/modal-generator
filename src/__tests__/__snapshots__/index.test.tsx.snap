// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Modal Check if modal is in the document 1`] = `
<body>
  <div />
  <div
    class="react-modal-container"
  >
    <div
      class="react-modal-content"
    >
      <button
        class="react-modal-close default"
      />
      modal content
    </div>
  </div>
</body>
`;

exports[`React Modal Check if modal is not in the document 1`] = `
<body>
  <div />
</body>
`;
